src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/PooledTxsRequestor.cs

```diff
@@ -94,30 +90,69 @@ public void RequestTransactionsEth68(Action<V66.Messages.GetPooledTransactionsMe
             RequestPooledTransactionsEth66(send, hashesToRequest);
         }
 
-        private ArrayPoolList<Hash256> AddMarkUnknownHashes(ReadOnlySpan<Hash256> hashes)
+        private ArrayPoolList<Hash256> AddMarkUnknownHashes(ReadOnlySpan<Hash256> hashes, Action<GetPooledTransactionsMessage> send, Guid sessionId)
+        {
+            ArrayPoolList<Hash256> discoveredTxHashes = new(hashes.Length);
+            for (int i = 0; i < hashes.Length; i++)
+            {
+                Hash256 hash = hashes[i];
+                if (!txPool.IsKnown(hash))
+                {
+                    if (txPool.RetryCache.Announced(hash, sessionId, () =>
+                    {
+                        ArrayPoolList<Hash256> hashesToRetry = new(1) { hash };
+                        GetPooledTransactionsMessage msg65 = new(hashesToRetry);
+                        send(msg65);
```

Comment:
We could avoid allocations of delegates, if we passed 2 separate actions. One for constructing the message and one for Send.

A bit less clean but no delegates allocated as the constructor one can be saved in private field (or be a method)

Reply:
Let's just get rid of send later?